src/algorithms/synthesis/encoding.cpp

```diff
@@ -15,7 +15,27 @@ namespace syrec {
             auto completeInputs = input.completeCubes();
             // move all the complete cubes to the new cube map
             for (auto const& completeInput: completeInputs) {
-                newTT.try_emplace(completeInput, output);
+                const auto inputIt = newTT.find(input);
+
+                if (inputIt != newTT.end()) {
+                    TruthTable::Cube newOutCube{newTT[input]};
```

Comment:
you already have an iterator to the element. No need to query it again with `newTT[input]`